odoo/addons/base/models/ir_actions.py

```diff
@@ -1170,57 +1170,62 @@ def run(self):
         """
         res = False
         for action in self.sudo():
-            action_groups = action.group_ids
-            if action_groups:
-                if not (action_groups & self.env.user.all_group_ids):
-                    raise AccessError(_("You don't have enough access rights to run this action."))
-            else:
-                model_name = action.model_id.model
-                try:
-                    self.env[model_name].check_access("write")
-                except AccessError:
-                    _logger.warning("Forbidden server action %r executed while the user %s does not have access to %s.",
-                        action.name, self.env.user.login, model_name,
-                    )
-                    raise
-
             eval_context = self._get_eval_context(action)
             records = eval_context.get('record') or eval_context['model']
             records |= eval_context.get('records') or eval_context['model']
-            self._can_execute_action_on_records(records)
-
-            if action.warning:
-                raise ServerActionWithWarningsError(_("Server action %(action_name)s has one or more warnings, address them first.", action_name=action.name))
+            action._can_execute_action_on_records(records)
+            res = action._run(records, eval_context)
+        return res or False
 
-            runner, multi = action._get_runner()
-            if runner and multi:
-                # call the multi method
-                run_self = action.with_context(eval_context['env'].context)
+    def _run(self, records, eval_context):
+        self.ensure_one()
+        if self.warning:
+            raise ServerActionWithWarningsError(_("Server action %(action_name)s has one or more warnings, address them first.", action_name=self.name))
```

Comment:
to do flch: check with expected override if correct to include raise if warning here instead of above.

So, BTW, misses link to ENT commit needing that refactor.